File: java/src/main/java/PaintCalculator.java
public class PaintCalculator {
    private static final double PAINT_FACTOR = 0.1;

    public double calculatePaintRequired(Rectangle rectangle) {
        return rectangle.area() * PAINT_FACTOR;
    }

    public double calculatePaintRequired(Square square) {
        return square.area() * PAINT_FACTOR;
    }

    public double calculatePaintRequired(Circle circle) {
        return circle.area() * PAINT_FACTOR;
    }
}

File: typescript/src/PaintCalculator.ts
import Rectangle from "./Rectangle";
import Square from "./Square";
import Circle from './Circle';

export default class PaintCalculator {

    static paintFactor = 0.1;

    static calculatePaintRequired(input: Circle | Square | Rectangle): number {
        return input.area() * PaintCalculator.paintFactor;
    }
}
